tests/test_api.py

```diff
@@ -730,3 +732,45 @@ def test_delete_build_auth(testclient):
 
     # r = schema.APIResponse.parse_obj(response.json())
     # assert r.status == schema.APIStatus.ERROR
+
+
+def test_api_cancel_build(testclient):
+    build_id = 1
+
+    testclient.login()
+    response = testclient.put(f"api/v1/build/{build_id}")
+    r = schema.APIPostSpecification.parse_obj(response.json())
+    assert r.status == schema.APIStatus.OK
+
+    new_build_id = r.data.build_id
+
+    # delay to ensure the build kicks off
+    building = False
+    while building is False:
```

Comment:
This while loop will never end is some cases. Need to have this as a timeout with check interval.

Reply:
Done, using `time`.  Good catch.